### PR TITLE
TransformControls: Add rotation snapping support

### PR DESCRIPTION
**Pull request by elisee**
This PR builds on #7158 (which added local space grid snapping). It renames `TransformControls.setSnap()` to `.setTranslationSnap()` and adds `.setRotationSnap()` which accepts a value in degrees for snapping rotations.

### CODE REVIEW

**Comment by mrdoob** [comment]
Looks good! I would use radians in `setRotationSnap()` though.

**Comment by elisee** [comment]
Updated.

Both `PerspectiveCamera` ([`fov`](https://github.com/mrdoob/three.js/blob/17df4c44c2365ad7649f53e99734ae48848ab646/src/cameras/PerspectiveCamera.js#L15)) and `EdgeHelpers` ([`thresholdAngle`](https://github.com/mrdoob/three.js/blob/17df4c44c2365ad7649f53e99734ae48848ab646/src/extras/helpers/EdgesHelper.js#L5)) use degrees, so I thought it made sense to use degrees here too, but I realize now most other places expect radians.

**Comment by mrdoob** [comment]
> Both `PerspectiveCamera` ([`fov`](https://github.com/mrdoob/three.js/blob/17df4c44c2365ad7649f53e99734ae48848ab646/src/cameras/PerspectiveCamera.js#L15)) and `EdgeHelpers` ([`thresholdAngle`](https://github.com/mrdoob/three.js/blob/17df4c44c2365ad7649f53e99734ae48848ab646/src/extras/helpers/EdgesHelper.js#L5)) use degrees, so I thought it made sense to use degrees here too, but I realize now most other places expect radians.

Uh... We'll have to address those eventually.

File: examples/js/controls/TransformControls.js [hunk 1]
```diff
@@ -617,7 +617,8 @@
 
 		this.object = undefined;
 		this.visible = false;
-		this.snap = null;
+		this.translationSnap = null;
```
Comment (mrdoob) [inline examples/js/controls/TransformControls.js]: We may want to make these variables private eventually too.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**